src/docs/authoring_extensions.md

```diff
@@ -160,75 +111,35 @@ export class HelloWorldMenuContribution implements MenuContribution {
 }
 ```
 
-## Executing the Extension In the Browser
+## Adding Extensions to a Theia application
 
-Now we want to see our extension in action. For this purpose, the generator has created a `package.json` in the folder `browser-app`. It defines a Theia browser application with a couple of statically included extensions, including our `hello-world-extension`. All remaining files in this directory have been auto-generated by `yarn` calling the `theia-cli` tool during the build, as defined in the scripts section.
+To make sure your extension is included in your Theia application, list it as a dependency in your browser or electron app, e.g. like this:
 
 ```json
 {
+  "private": true,
   "name": "browser-app",
-  "version": "0.1.0",
+  "version": "0.0.0",
   "dependencies": {
     "@theia/core": "latest",
-    "@theia/filesystem": "latest",
-    "@theia/workspace": "latest",
-    "@theia/preferences": "latest",
-    "@theia/navigator": "latest",
-    "@theia/process": "latest",
-    "@theia/terminal": "latest",
-    "@theia/editor": "latest",
-    "@theia/languages": "latest",
-    "@theia/markers": "latest",
-    "@theia/monaco": "latest",
-    "@theia/messages": "latest",
-    "hello-world-extension": "0.1.0"
+    ...
+    "hello-world": "0.0.0"
   },
   "devDependencies": {
     "@theia/cli": "latest"
   },
   "scripts": {
-    "prepare": "theia build",
+    "bundle": "yarn rebuild && theia build --mode development",
+    "rebuild": "theia rebuild:browser --cacheRoot ..",
     "start": "theia start",
-    "watch": "theia build --watch"
+    "watch": "yarn rebuild && theia build --watch --mode development"
   },
   "theia": {
     "target": "browser"
   }
 }
 ```
 
-Now we have all pieces together to build and run the application.
-To run the browser app, enter:
-
-```bash
-cd browser-app
-yarn start <path to workspace>
-```
-
-Point your browser to <http://localhost:3000>. Then choose _Edit > Say Hello_ from the menu: A message "Hello World!" should pop up.
-
-## Executing the Extension In Electron
-
-The `package.json` for the Electron app looks almost the same, except for the name and the target property.
-
-```json
-{
-  "name": "electron-app",
-  ...
-  "theia": {
-    "target": "electron"
-  }
-}
-```
-
-Before running the electron app, you additionally have to rebuild some native modules:
-
-```bash
-yarn rebuild:electron
-cd electron-app
-yarn start <path to workspace>
-```
-
 ## Deploying the Extension
```

Comment:
I find the section on Deploying the Extension now a bit out of place. Maybe we change it to:

## Running the Extension

To run the extension, you have two options:
1. Have your extension as part of a monorepo containing a Theia-based appllication importing your extension
2. Publish the extension with `yarn publish` and consume it from your Theia-bsed application

See [*Executing the Browser Application*](/docs/composing_applications/#executing-the-browser-application) and [*Executing the Extension in Electron* for more details](/docs/composing_applications/#executing-the-extension-in-electron) for adding extensions to the dependencies of a Theia-based application and running it.